### DISCUSSION TITLE
Print statements in ASGI middleware never prints.

### DISCUSSION THREAD

**Discussion by rajaravivarma-r** [fastapi/fastapi#9540] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import logging
import asyncio

from fastapi import FastAPI
from fastapi.responses import JSONResponse

class LogRequestResponseMiddleware:
    def __init__(self, app):
        self.app = app

    async def __call__(self, scope, receive, send):
        if scope["type"] != "http":
            await self.app(scope, receive, send)
            return

        async def inner_receive():
            print("Inner receive: awaiting outer receive") # Not working
            message = await receive()
            print("Inner receive: got message") # Not working

            return message

        async def inner_send(message):
            print("Inner send: before") # Working
            await send(message)
            print("Inner send: after") # Working

        try:
            await self.app(scope, inner_receive, inner_send)
        except BaseException as exc:
            print(str(exc))

# minimal_main
app = FastAPI(debug=True)

app.add_middleware(LogRequestResponseMiddleware)

@app.post('/root')
async def root():
    return JSONResponse(dict(name="raja", country="India"))


# Executed like this `uvicorn --log-level=debug --reload minimal_main:app`
```

### Description

* Execute the command `curl -X POST 'http://localhost:8000/root'  --data-raw '{"name": "tony"}'`
* Expected:
  * Print `Inner receive: awaiting outer receive` to stdout
  * Print `Inner receive: got message` to stdout
 * Observed:
   * The text statements were never printed.

### Operating System

macOS

### FastAPI Version

0.95.1

### Python Version

3.11.3

### Additional Context

The behaviour seems to be consistent with both `uvicorn` and `hypercorn`.

In fact, the `inner_receive` never seems to be executed at all. But, on the other hand, the `inner_send` is getting called and works as expected.

I have also tried `logger` instead of `print`, but logger doesn't work either.

I'm using this to peek into the request body.

**Comment by kumarswamy-itt** [comment]
I am facing the same issue @rajaravivarma-r did you get any solution?

**Comment by rajaravivarma-r** [comment]
@kumarswamy-itt I sort of managed to log things to a log file. If you are interested, then here is a sample script I posted https://gist.github.com/rajaravivarma-r/bb9bd10e9c21084410918061d113582c

**Reply by lz-work** [reply]
Same and better try logging instead. Well here is some inspiration https://github.com/tiangolo/fastapi/discussions/11284 I hope.
